crates/optimism/txpool/src/supervisor/client.rs

```diff
@@ -138,14 +143,16 @@ impl CheckAccessListRequest<'_> {
     }
 }
 
-impl<'a> IntoFuture for CheckAccessListRequest<'a> {
+impl<'a, 'b: 'a> IntoFuture for CheckAccessListRequest<'a, 'b> {
     type Output = Result<(), InteropTxValidatorError>;
     type IntoFuture = BoxFuture<'a, Self::Output>;
 
     fn into_future(self) -> Self::IntoFuture {
-        let Self { client, inbox_entries, executing_descriptor, timeout, safety } = self;
+        let Self { client, inbox_entries, executing_descriptor, timeout, safety, metrics } = self;
         Box::pin(async move {
-            tokio::time::timeout(
+            let start = Instant::now();
```

Comment:
this is ok, but let's be as precise with timing operations as possible as good practice 
```rust
let start = Instant::now();

let result = tokio::time::timeout(
    timeout,
    client.request(
        "supervisor_checkAccessList",
        (inbox_entries, safety, executing_descriptor),
    ),
)
.await;
            
metrics.record_supervisor_query(start.elapsed());
            
result .map_err(|_| InteropTxValidatorError::ValidationTimeout(timeout.as_secs()))?
        .map_err(InteropTxValidatorError::client)
        .map_err(InteropTxValidatorError::client)
```